src/geometry.F90

```diff
@@ -475,6 +475,68 @@ subroutine cross_surface(p, last_cell)
              &// trim(to_str(surf%id)))
       end if
       return
+    elseif (surf % bc == BC_PERIODIC .and. run_mode /= MODE_PLOTTING) then
+      ! =======================================================================
+      ! PERIODIC BOUNDARY
+
+      ! Do not handle periodic boundary conditions on lower universes
+      if (p % n_coord /= 1) then
+        call handle_lost_particle(p, "Cannot period particle " &
+             // trim(to_str(p % id)) // " off surface in a lower universe.")
+        return
+      end if
+
+      ! Score surface currents since reflection causes the direction of the
+      ! particle to change -- artificially move the particle slightly back in
+      ! case the surface crossing is coincident with a mesh boundary
+
+      if (active_current_tallies % size() > 0) then
+        p % coord(1) % xyz = p % coord(1) % xyz - TINY_BIT * p % coord(1) % uvw
+        call score_surface_current(p)
+        p % coord(1) % xyz = p % coord(1) % xyz + TINY_BIT * p % coord(1) % uvw
```

Comment:
Rather than subtracting and adding `TINY_BIT`, I suggest we instead save the `xyz` to a local variable and set `xyz` back to the same exact value.  Actually, I wonder if this code for reflective BCs is why I still see lost particles on big simulations.